iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorNonLinearModelImpl.java

```diff
@@ -0,0 +1,142 @@
+            return this;
+        }
+
+        @Override
+        public double getG() {
+            return g;
+        }
+
+        @Override
+        public Section setG(double g) {
+            ValidationUtil.checkG(shuntCompensator, g);
+            double oldValue = this.g;
+            this.g = g;
+            shuntCompensator.notifyUpdate(notifyUpdateSection(index, "g"), oldValue, this.g);
+            return this;
+        }
+
+        private static String notifyUpdateSection(int sectionNum, String attribute) {
+            return "section" + sectionNum + "." + attribute;
+        }
+
+        void setShuntCompensator(ShuntCompensatorImpl shuntCompensator) {
+            if (this.shuntCompensator != null) {
+                throw new PowsyblException("Shunt compensator " + shuntCompensator.getId() + " has been set twice for the section " + index);
+            }
+            this.shuntCompensator = shuntCompensator;
+        }
+    }
+
+    private final List<SectionImpl> sections;
+
+    ShuntCompensatorNonLinearModelImpl(List<SectionImpl> sections) {
+        this.sections = sections;
+    }
+
+    @Override
+    public Section getSection(int sectionIndex) {
+        if (sectionIndex < 1 || sectionIndex > sections.size()) {
+            throw new ValidationException(shuntCompensator, "invalid section index (must be in [1;maximumSectionCount]");
+        }
+        return sections.get(sectionIndex - 1);
+    }
+
+    @Override
+    public List<Section> getAllSections() {
+        return Collections.unmodifiableList(sections);
+    }
+
+    @Override
+    public ShuntCompensatorModelType getType() {
+        return ShuntCompensatorModelType.NON_LINEAR;
+    }
+
+    @Override
+    public int getMaximumSectionCount() {
+        return sections.size();
+    }
+
+    @Override
+    public double getB(int sectionCount) {
+        if (sectionCount < 0 || sectionCount > sections.size()) {
+            throw new ValidationException(shuntCompensator, "invalid section count (must be in [0;maximumSectionCount]");
+        }
+        return sectionCount == 0 ? 0 : sections.get(sectionCount - 1).getB();
+    }
+
+    @Override
+    public double getG(int sectionCount) {
+        if (sectionCount < 0 || sectionCount > sections.size()) {
+            throw new ValidationException(shuntCompensator, "invalid section count (must be in [0;maximumSectionCount]");
+        }
+        return sectionCount == 0 || Double.isNaN(sections.get(sectionCount - 1).getG()) ? 0 : sections.get(sectionCount - 1).getG();
+    }
+
+    @Override
+    public double getBPerSection(int sectionIndex) {
+        if (sectionIndex < 1 || sectionIndex > sections.size()) {
+            throw new ValidationException(shuntCompensator, "invalid section index (must be in [1;maximumSectionCount]");
+        }
+        return sectionIndex == 1 ? sections.get(0).getB() : sections.get(sectionIndex - 1).getB() - sections.get(sectionIndex - 2).getB();
```

Comment:
This is very dirty code just because of index > 0 convention ?

Reply:
Method deleted (as every method using `sectionIndex` concept)